addons/account/models/account_move.py

```diff
@@ -3578,3 +3579,29 @@ def _get_update_invoice_from_attachment_decoders(self, invoice):
         :returns:       A list of tuples (priority, method) where method takes an attachment as parameter.
         """
         return []
+
+    @api.model
+    def get_required_fields_for_country(self, country_id):
+        """ Returns the list of fields that needs to be filled when creating an invoice for the selected country.
+        This is required for some flows that would allow a user to request an invoice from the portal.
+
+        :param country_id: The country for which we want the fields.
+        :return: an array of fields names for which the user should provide values. Html fields will be generated for them.
+        """
+        return []
+
+    @api.model
+    def _field_to_html(self, field_name):
+        """ Simple helper to transform fields into basic html form element.
+        """
+        field = self._fields[field_name]
+        html = ''
+        if field.type == 'selection':
+            desc = [f'<select class="form-select" name="{field.name}" required="required">']
+            selections = field._description_selection(self.env)
+            desc.append(f'<option selected="true" disabled="disabled" value="">Select the {field.string.lower()}...</option>')
+            for value, label in selections:
+                desc.append(f'<option value="{value}">{label}</option>')
+            desc.append('</select>')
+            html = Markup(''.join(desc))
+        return html
```

Comment:
@odoo/rd-security Hi team, could I get a review for this please :pray: 
The purpose is to be able to dynamically make HTML fields from model fields in a simple manner to display a small form to the user in the portal.
I preferred to do the HTML here so that it's easier to correct and/or add new fields if really needed without having to update an existing template.

Let me know if it's ok or if we already have this somewhere, I couldn't really find it :smile:

Reply:
You have to move the variable parts out of the Markup call to be properly escaped
```py
>>> value = "42"
>>> label = "broken </option>"
>>> Markup(f"""<select><option value="{value}>">{label}</option></select>""")
Markup('<select><option value="42>">broken </option></option></select>')
>>> Markup("""<select><option value="{value}>">{label}</option></select>""").format(value=value, label=label)
Markup('<select><option value="42>">broken &lt;/option&gt;</option></select>')
```

Reply:
Thank you :+1: I changed it, please let me know if this way is ok thanks :smile:

Reply:
Where is it used anyway? :thinking:

Reply:
In enterprise https://github.com/odoo/enterprise/pull/30599/files#diff-8afc82e663ecf1ebc79d464ba7d9e96a7fda9fe2b431abecbd4f7216114de2dfR532
Here because it's needed only for now when using mx edi, but it will most likely be needed later for other l10n and not necessarily in enterprise so I put it here directly.

Reply:
Oh I see. If you don't have any other PR using it planned, I think it's better to move it in l10n_mx_edi at the moment and move it to `account` [once we actually need it](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it).